### DISCUSSION TITLE
Auto Associate a Custom Object field to a Project via the Task that field is on

### DISCUSSION THREAD

**Discussion by hagen31** [sadieblue/agilityblue.workspace-scripts#36] [Q&A]
On a Task I have a field called Collection Log that is a Reference field to a Custom Object of the same name. When a user selects the field they get a box to associate to an existing entry; however, normally they need to create the entry so they do which brings up the options for Collection Log Object. The first field is called Collection Request that is a Reference field to the Projects Object.

What I would like to happen is for this field to automatically populate with the Project that is associated to the Task where they started. It seems to me this is four "hops back" as.

1. User opens the Task - this holds the ProjectID
2.  User selects the Collection Log field
3.  User selects Create New
4. This is where user needs to associate to the ProjectID

Is it possible to script this automation?

**Comment by bmward80** [comment]
Hi Austin,

Scripts can be used in events where data is being saved in some way, such as something being created, updated, or deleted. In the scenario you presented, nothing has been saved, so there is no where for a script to be attached.

Since PowerShell scripting is all API-driven on the server side, it wouldn't work with being able to do anything on the browser/UI side like you're describing. For being able to automate something like that, we would need to create a UI-driven scripting system that could support custom JavaScript.

Brandon

**Comment by hagen31** [comment]
Hello Brandon,

I’m not sure I’m following as the final step would be to save the collection log which then links it to that project and task.


[REDACTED SIGNATURE INFORMATION]

**Reply by bmward80** [reply]
Hi Austin,

I think I understand. In the scenario you presented, it seemed like as the user is creating a new collection log entry, it should automatically populate a project field as they are creating the entry, but I think you are asking to have it populate after the save, where scripting could come into play.

The source of the "hops" is not recorded anywhere, so I'm not sure you'd be able to do that. When you save the new entry, as far the server is aware, there is no understanding of how you got to the point, all it knows is that you're creating a new entry. The script would need to somehow know how to associate it to the correct project. That would be either using some kind of logic that's obvious enough for you to to script out based on the fields of the entry, or some kind of reference like an ID or name or something.

Brandon

**Comment by hagen31** [comment]
Hello Brandon,

Before or after save either one works; however, it sounds like I will not be able to do that. I was hoping the system recorded the Project ID as it would save a user clicks and this process already requires a lot of clicks.

Thank you,

Austin J. Hagen | Proteus Discovery Group
Vice President, Operations
m: 317.989.8314
Rice Lake, WI
***@***.******@***.***>
www.proteusdiscovery.com<http://www.proteusdiscovery.com/>

**Comment by hagen31** [comment]
Closing as what I was looking to do is not possible.
